frontend/src/scenes/notebooks/Nodes/NotebookNodeQuery.tsx

```diff
@@ -70,8 +89,15 @@ export const Settings = ({
     attributes,
     updateAttributes,
 }: NotebookNodeAttributeProperties<NotebookNodeQueryAttributes>): JSX.Element => {
+    // insightLogic
+    const logic = insightLogic({
+        dashboardItemId: (attributes.query as SavedInsightNode).shortId || 'new',
```

Comment:
Does this cause issues if multiple insights co-exist in a notebook? I think the shared key of `new` will mean that the callbacks may fire incorrectly

Reply:
If we have multiple `new` insights within a notebook, then, yeah. We'll likely need some form of new keying for these